### PR TITLE
Display node name in performance profiling output (instrumentation) 

### PR DESCRIPTION
**Pull request by imaihal**
This PR adds node name in profile output. The node name is displayed on the right size as shown in following example(from yolov3.onnx). This is useful to identify nodes which take long time. If the op don't have node name attribute, it is not displayed.

```
  :
#892) before op=  Return Time elapsed: 0.000000 accumulated: 1664933967.481642
#893) before op=     Add Time elapsed: 0.000001 accumulated: 1664933967.481643 (add__252)
#894) after  op=     Add Time elapsed: 0.000000 accumulated: 1664933967.481643 (add__252)
#895) before op=  Return Time elapsed: 0.000002 accumulated: 1664933967.481645
#896) after  op=    Loop Time elapsed: 0.000000 accumulated: 1664933967.481645 (TFNodes/yolo_evaluation_layer_1/arange_1__246_loop)
#897) before op= Reshape Time elapsed: 0.000001 accumulated: 1664933967.481646 (TFNodes/yolo_evaluation_layer_1/Reshape_2)
#898) after  op= Reshape Time elapsed: 0.000000 accumulated: 1664933967.481646 (TFNodes/yolo_evaluation_layer_1/Reshape_2)
#899) before op= Constan Time elapsed: 0.000001 accumulated: 1664933967.481647
#900) after  op= Constan Time elapsed: 0.000000 accumulated: 1664933967.481647
#901) before op=    Ceil Time elapsed: 0.000001 accumulated: 1664933967.481648 (TFNodes/yolo_evaluation_layer_1/arange__271_ceilTFNodes/yolo_evaluation_layer_1/Reshape_2)
#902) after  op=    Ceil Time elapsed: 0.000001 accumulated: 1664933967.481649 (TFNodes/yolo_evaluation_layer_1/arange__271_ceilTFNodes/yolo_evaluation_layer_1/Reshape_2)
#903) before op=    Cast Time elapsed: 0.000000 accumulated: 1664933967.481649 (TFNodes/yolo_evaluation_layer_1/arange__271_trip_cnt)
#904) after  op=    Cast Time elapsed: 0.000001 accumulated: 1664933967.481650 (TFNodes/yolo_evaluation_layer_1/arange__271_trip_cnt)
#905) before op=    Loop Time elapsed: 0.000001 accumulated: 1664933967.481651 (TFNodes/yolo_evaluation_layer_1/arange__271_loopTFNodes/yolo_evaluation_layer_1/arange__271_trip_cnt)
#906) before op=     Add Time elapsed: 0.000006 accumulated: 1664933967.481657 (add__277)
#907) after  op=     Add Time elapsed: 0.000000 accumulated: 1664933967.481657 (add__277)
   :
```

### CODE REVIEW

**Comment by AlexandreEichenberger** [comment]
@chentong319 do you mind looking into this, it looks like a simple change to the code you wrote.

File: src/Dialect/Krnl/KrnlOps.cpp [hunk 1]
```diff
@@ -511,11 +511,15 @@ void KrnlInstrumentOp::build(mlir::OpBuilder &builder, OperationState &state,
...
-  state.addAttribute("opName", nameAttr);
+  StringAttr nodeNameAttr =
+      op->getAttrOfType<::mlir::StringAttr>("onnx_node_name");
+  IntegerAttr tagAttr = builder.getI64IntegerAttr(tag);
+  StringAttr opNameAttr = builder.getStringAttr(StringRef(opName));
+  state.addAttribute("opName", opNameAttr);
   state.addAttribute("opID", attr);
   state.addAttribute("tag", tagAttr);
+  if (nodeNameAttr)
+    state.addAttribute("nodeName", nodeNameAttr);
```
Comment (chentong319) [inline src/Dialect/Krnl/KrnlOps.cpp:522]: Can we use default builder instead of addAttribute here?

**Review by chentong319 — APPROVED** [review]
LGTM!

File: src/Dialect/Krnl/KrnlOps.cpp (hunk 1 above)
Comment (imaihal) [reply src/Dialect/Krnl/KrnlOps.cpp]: @chentong319 I updated to  use the default builder. Thanks!

**Comment by chentong319** [comment]
> @chentong319 Thanks for your review. I updated docs.
> 
> Let me check just to be sure. This PR changed to show only the `after` line since I always use both `--InstrumentBeforeOp` and `--InstrumentAfterOp` and check only `after` line. Is this change acceptable? Is there any usecase we wanted to check `before` line?

Using both `before` and `after` can tell the time spent on the operations added between onnx ops. For example, the end of LoopOp body may contain some operations to accumulate loop body result. `before` line can also be used to set break point. 
Do not remove it.

**Comment by imaihal** [comment]
> For example, the end of LoopOp body may contain some operations to accumulate loop body result. `before` line can also be used to set break point.

I see. I will update/revert some commit. Thanks.

**Comment by imaihal** [comment]
@chentong319 Now this PR simply adds node name in profile output. Could you review again?

**Review by chentong319 — APPROVED** [review]
LGTM!

**Pull request merged**